### ISSUE TITLE
Possible to add RAG and small GUI?

### ISSUE THREAD

**Issue by fxtoofaan** [joaomdmoura/crewAI#18]
Would love to see some kind of RAG and UI implemented. Thank you

**Comment by hugofreire** [comment]
RAG will be awesome

**Comment by fxtoofaan** [comment]
local RAG and local ollama llm model(s) example code please :)

**Comment by joaomdmoura** [comment]
Nice, okay I hear you, I'll look into a specific simple RAG default tool maybe and add more ollama examples

**Comment by okoliechykwuka** [comment]
Thanks, @joaomdmoura , I noticed you implemented RAG in the financial analyst example. 

https://youtu.be/e0Uj4yWdaAg

**Comment by mindwellsolutions** [comment]
@joaomdmoura Thank you for working on adding RAG, this will be huge. The ability for CrewAI to read through a whole folder of  documents, pdfs (potentially spreadsheets, ebooks, datasets) as its research source, in place of DuckDuckGoSearch using the internet for research. This would allow people to point the agent towards larger bodies of knowledge (pdfs) and books, to help it research and figure out its responses. This could be incredibly powerful :)

**Comment by EmmaWebGH** [comment]
+1

I would really, really like to be able to 
a) give agents bodies of specific knowledge via RAG

And 
b) also give them examples of their expected output as docs they can reference.

Maybe (b) can be achieved in prompt, but this would be clunky I feel.
Or (b) maybe if we're using ChatGPT4.x, we can feed it a document and hope it understands the format enough to mimic on output.

But definitely (a) giving them specific knowledge would be the next logical step to empowering the agents to do different jobs?

**Comment by joaomdmoura** [comment]
This is getting prioritized to be worked on next

**Comment by EmmaWebGH** [comment]
Excellent news! Very excited. 

I think this would really supercharge crewAI and give it the best of both worlds that Autogen (agents) and taskweaver (code-first) have. One of the main things I would love to try is mixing general knowledge with coding ability, and with Autogen/Taskweaver I feel you get one or the other - not both!

**Comment by WismutHansen** [comment]
Would this help with implementing RAG: https://github.com/KillianLucas/aifs

**Comment by scaruslooner** [comment]
excited

**Comment by slavakurilyak** [comment]
[AI File System](https://github.com/KillianLucas/aifs) can be used as a tool for crewAI that would unlock RAG-pipelines for local search (file or folder search)

**Comment by slavakurilyak** [comment]
[RAG](https://github.com/joaomdmoura/crewai-tools/tree/main/crewai_tools/tools/rag) is now part of the [crewai-tools](https://github.com/joaomdmoura/crewai-tools) library

**Comment by slavakurilyak** [comment]
Excited to see the new UI

**Comment by bhargav-11** [comment]
bump!!!

**Comment by airtonix** [comment]
don't waste time on a UI. it's orthogonal to the purpose of this library.

**Comment by joaomdmoura** [comment]
Yup UI us coming, we kept on pushing it to build the backbones of the lib but we will also need it ourselves, so expected it sooner than later

**Comment by joaomdmoura** [comment]
https://github.com/joaomdmoura/crewAI/issues/18#issuecomment-1955750679
About RAG ^

**Comment by joaomdmoura** [comment]
UI is in the works and scheduled to drop in the next 30d :)
Rag is now supported over tools :D
--
João Moura
@joaomdmoura <http://twitter.com/joaomdmoura>


Em ter., 30 de abr. de 2024 às 18:19, Maged Helmy ***@***.***>
escreveu:

**Comment by shawnholt** [comment]
A lot of people have been drawn to autogen because it's much more comfortable to work with a GUI , rather than code. I see a lot of abandoned attempts and "coming soon", but based on this thread alone it does not seem like it is a priority, please help us understand where it fits into the roadmap. Thank you!

**Comment by cbarkinozer** [comment]
Is there any news or an ETA from UI?
